### PR TITLE
DIM-111: Fully working ROSCommandQueue orchestration between nav2 and webrtc action primitives

### PR DESCRIPTION
**Pull request by spomichter**
Current working implementation in run_go2_ros.py

Likely should do more testing for edge cases

### CODE REVIEW

File: dimos/robot/ros_command_queue.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+                    self._logger.debug(f"Robot busy state changed: {self._last_busy_state} -> {is_busy}")
+                    self._last_busy_state = is_busy
+                    
+                    # If the robot has transitioned to busy, record the time
+                    if is_busy:
+                        self._stuck_in_busy_since = current_time
+                    else:
+                        self._stuck_in_busy_since = None
+                
+                # Check if we've been waiting too long for the robot to be ready
```
Comment (lukasapaukstys) [inline dimos/robot/ros_command_queue.py:289]: It appears that forcing queue continuation after a prolonged busy state might be a workaround for an underlying race condition or scheduling issue. We should investigate why the robot remains ‘busy’ for extended periods and address the root cause. Additionally, revisiting the connection/disconnection states and potentially integrating a connection manager might help prevent these issues. We can revisit this at a later point.
